src/PKSim.Core/Mappers/ExpressionProfileToExpressionProfileBuildingBlockMapper.cs

```diff
@@ -0,0 +1,77 @@
+﻿using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility;
+using PKSim.Core.Model;
+using IFormulaFactory = PKSim.Core.Model.IFormulaFactory;
+
+namespace PKSim.Core.Mappers
+{
+   public interface IExpressionProfileToExpressionProfileBuildingBlockMapper : IMapper<ExpressionProfile, ExpressionProfileBuildingBlock>
+   {
+   }
+
+   public class ExpressionProfileToExpressionProfileBuildingBlockMapper : IExpressionProfileToExpressionProfileBuildingBlockMapper
+   {
+      private readonly IObjectBaseFactory _objectBaseFactory;
+      private readonly IEntityPathResolver _entityPathResolver;
+      private IFormulaFactory _formulaFactory;
+
+      public ExpressionProfileToExpressionProfileBuildingBlockMapper(IObjectBaseFactory objectBaseFactory, IEntityPathResolver entityPathResolver, IFormulaFactory formulaFactory)
+      {
+         _objectBaseFactory = objectBaseFactory;
+         _entityPathResolver = entityPathResolver;
+         _formulaFactory = formulaFactory;
+      }
+
+      public ExpressionProfileBuildingBlock MapFrom(ExpressionProfile expressionProfile)
+      {
+         var expressionProfileBuildingBlock = _objectBaseFactory.Create<ExpressionProfileBuildingBlock>();
+
+         expressionProfileBuildingBlock.Name = expressionProfile.Name;
+         expressionProfileBuildingBlock.PKSimVersion = ProjectVersions.Current;
+         expressionProfileBuildingBlock.Description = expressionProfile.Description;
+
+         var moleculeType = expressionProfile.Molecule.MoleculeType;
+         switch (moleculeType)
+         {
+            case QuantityType.Enzyme:
+               expressionProfileBuildingBlock.Type = ExpressionTypes.MetabolizingEnzyme;
+               break;
+            case QuantityType.Transporter:
+               expressionProfileBuildingBlock.Type = ExpressionTypes.TransportProtein;
+               break;
+            default:
+               expressionProfileBuildingBlock.Type = ExpressionTypes.ProteinBindingPartner;
+               break;
+         }
+
+         var allParameters = expressionProfile.GetAllChildren<IParameter>();
+
+         foreach (var parameter in allParameters)
+         {
+            var expressionParameter = _objectBaseFactory.Create<ExpressionParameter>();
```

Comment:
can we have a private method here that justs maps a parameter. Thsi is a fairly simple method but it's quite long and I think they are are clear sections that can be moved  out

1/ the ExpressionTyupes to expressionProfileBuildingBlock.Type (BTW, any reason why we have different types and not the same?) (so line 36 to 48 can be a method with simple return statements

Like 54 to 70 is also a mapper method so to speak. 

Thoughts?

Reply:
done at least those, if you want we can discuss